src/utils/__tests__/api/helpers/ecommerce/ecommercePaymentMethodsHelper.test.ts

```diff
@@ -200,6 +201,21 @@ describe("Ecommerce payment methods helper - calculateFees tests", () => {
 });
 
 describe("Ecommerce payment methods helper - getPaymentInstruments tests", () => {
+  it("Should call onError with GENERIC_ERROR and onResponse with empty array when API returns status 500", async () => {
+    // Simula presenza di token in sessione (quindi ramo V3)
+    (getSessionItem as jest.Mock).mockReturnValue("fake-auth-token");
+
+    // Simula che getAllPaymentMethodsV3 ritorni un Either.left
+    (
+      apiPaymentEcommerceClientV3.getAllPaymentMethodsV3 as jest.Mock
+    ).mockReturnValue(Promise.resolve(E.left(new Error("some error"))));
+
+    await getPaymentInstruments({ amount: 100 }, mockOnError, mockOnResponse);
+
+    expect(mockOnError).toHaveBeenCalledWith(ErrorsType.GENERIC_ERROR);
+    expect(mockOnResponse).toHaveBeenCalledWith([]); // perché ritorna []
```

Comment:
```suggestion
    expect(mockOnResponse).toHaveBeenCalledWith([]); // returns empty array on error
```
English comment (but I think we can also delete this comment, it's clear from the context)